Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -333,6 +358,35 @@ void UPSWorldSubsystem::OnSpotComponentLoad(UPSSpotComponent* SpotComponent)
 	PSCurrentSpotComponentInternal = SpotComponent;
 }
 
+// Removes progression system spot that should not be available by other objects anymore.
+void UPSWorldSubsystem::RemoveProgressionSystemSpot(UPSSpotComponent* ProgressionSystemSpotComponent)
+{
+	if (ensureMsgf(ProgressionSystemSpotComponent, TEXT("%s: 'ProgressionSystemSpotComponent' is null"), *FString(__FUNCTION__)))
+	{
+		PSSpotComponentArrayInternal.RemoveSwap(ProgressionSystemSpotComponent);
+	}
+}
+
+// Destroy all star actors that should not be available by other objects anymore.
+void UPSWorldSubsystem::DestroyStarActors()
```

Comment:
Please refactor your cleanup methods to make it more transperent for others what is cleaned

### DestroyStarActors
In `DestroyStarActors`, others will expect you only cleanup Stars, nothing else
```
	UPoolManagerSubsystem::Get().EmptyPool(UPSDataAsset::Get().GetStarActorClass());
	PoolActorHandlersInternal.Empty();
	ProgressionSettingsDataInternal.Empty();
	StarDynamicProgressMaterial = nullptr;
```

### Saves
Wrap Saves-related code to own function as well

### OnDeinitializeCleanup
Create new `OnDeinitializeCleanup`, where you will call all other methods, example:
```
        DestroyStarActors();
        
        PSDataAssetInternal.Reset();
	PSHUDComponentInternal = nullptr;
	PSSpotComponentArrayInternal.Empty();
	PSCurrentSpotComponentInternal = nullptr;

        UnloadSaves();
```

### Proper calling `OnDeinitializeCleanup`
Call `OnDeinitializeCleanup` from two places: OnUnloading (instead of DestroyStarActors), and from Deinitialize's engine virtual method of the world subsystem

Reply:
Since the DestroyAllStarActors is quite small function, I propose to go with comments inside the function instead of creating new method. 
So I renamed DestroyStarActors to PerformCleanUp and added logical comments section.
let me know if this works https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/6959e89039c62706b9e0772008e4c7aa09a310df